complete readme redesign

vitormmatos opened the pull request:
updated deno description matching the official site.

deno logo added aligned to the right side of the description.

twitter badge added.

installation, getting started and contributing sections have been added

<!--
Before submitting a PR, please read
https://github.com/denoland/deno/blob/master/docs/contributing.md
-->
<img width="887" alt="readme-update-screenshot" src="https://user-images.githubusercontent.com/14911617/93406422-00ae5880-f866-11ea-88fd-064f3f8883d7.png">

piscisaureus committed:
Revert "feat(install): bundle before installation" (#7522)

This reverts the changes introduced by PR #5276, which made
`deno install «script»` automatically bundle the script's dependencies.
It broke the `deno install` command for a large number of scripts.

This reverts commit 34e98fa59cd70f7ce64e587bef41fac536a3076b.

Closes: #7492

ry committed:
simplify global properties (#7502)

ry committed:
Remove unnecessary serde_derive dependency

ry committed:
Remove unnecessary extern statements

ry committed:
Re-export deno_core::url (#7525)

Also re-exports deno_core::futures and deno_core::serde_json but these are not yet used in the CLI.

bartlomieju committed:
refactor: remove dispatch_json.js from cli/rt and cli/tsc (#7521)

Instead use Deno.core.jsonOpSync and Deno.core.jsonOpAsync

caspervonb committed:
test(std/wasi): add sched_yield test (#7511)

Commit:
complete readme redesign

updated deno description matching the official site.

deno logo added aligned to the right side of the description.

twitter badge added.

installation, getting started and contributing sections have been added

Commit:
complete readme redesign

updated deno description matching the official site.

deno logo added aligned to the right side of the description.

twitter badge added.

installation, getting started and contributing sections have been added

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/denoland/deno?pullRequest=7530) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno?pullRequest=7530) before we can accept your contribution.<br/>**7** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: piscisaureus<br/>:white_check_mark: ry<br/>:white_check_mark: nayeemrmn<br/>:white_check_mark: lucacasonato<br/>:white_check_mark: caspervonb<br/>:white_check_mark: g-plane<br/>:white_check_mark: bartlomieju<br/>:x: Vitor Matos<br/>:x: JayHelton<br/>:x: tokiedokie<br/><hr/>**Vitor Matos** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno?pullRequest=7530) it.</sub>

kitsonk commented:
It is probably a really good idea to discuss these sorts of changes with someone on the core team _before_ you attempt to raise a PR.

ry commented:
@kitsonk is right - but in this case I like it! Please sign the CLA and I'll land.

tokiedokie commented on README.md:
```diff
@@ -1,29 +1,86 @@
+
+You can find a more in depth introduction, examples, and environment setup guides in the [manual](https://deno.land/manual).
+
+More in-depth info can be found in the runtime [documentation](doc.deno.land)
```
```suggestion
More in-depth info can be found in the [runtime documentation](https://doc.deno.land/builtin/stable)
```

tokiedokie replied:
missing url

nayeemrmn replied:
```suggestion
More info can be found in the [runtime documentation](https://doc.deno.land/builtin/stable).
```

tokiedokie commented on README.md:
```diff
@@ -1,29 +1,86 @@
+
+We appreciate your help!
+
+To contribute, please read the our [guidelines](https://github.com/denoland/deno/blob/master/docs/contributing/style_guide.md)
```
```suggestion
To contribute, please read the our [guidelines](https://deno.land/manual/contributing)
```

tokiedokie replied:
I think we should use deno.land/manual instead of a markdown file

nayeemrmn replied:
```suggestion
To contribute, please read our [guidelines](https://deno.land/manual/contributing).
```

tokiedokie commented on README.md:
```diff
@@ -1,29 +1,86 @@
+
+See  [deno_install](https://github.com/denoland/deno_install/blob/master/README.md) and [releases](https://github.com/denoland/deno/releases) for other options.
+
+### Getting Started
```
```suggestion
## Getting Started
```

tokiedokie commented on README.md:
```diff
@@ -1,29 +1,86 @@
+- Set of reviewed standard modules that are guaranteed to work with [Deno](https://deno.land/std/)
 
-[Install](https://github.com/denoland/deno_install)
+### Install
```
```suggestion
## Install
```

tokiedokie commented on README.md:
```diff
@@ -1,29 +1,86 @@
+Deno is a *simple*, *modern* and *secure* runtime for **JavaScript** and **TypeScript** that uses V8 and is built in Rust.
 
-[Website](https://deno.land/)
+### Features
```
```suggestion
## Features
```

nayeemrmn commented on README.md:
```diff
@@ -1,29 +1,86 @@
+- Supports TypeScript out of the box.
+- Ships only a single executable file.
+- Built-in utilities like a dependency inspector (deno info) and a code formatter (deno fmt).
+- Set of reviewed standard modules that are guaranteed to work with [Deno](https://deno.land/std/)
```
```suggestion
- Set of reviewed standard modules that are guaranteed to work with [Deno](https://deno.land/std/).
```

vitormmatos commented:
> @kitsonk is right - but in this case I like it! Please sign the CLA and I'll land.

Just signed the CLA. 
I know it's not much, but I hope I can help the project, now and in the future.
About discussing with the core team for future pull requests, where can it be done?

Thanks for the support.

kitsonk commented:
@vitormmatos two main ways, an issue stating what the problem is and the suggested approach to fix it, second would be on [Discord](https://discord.gg/deno) in the #dev channel there.

ry committed:
refactor: make fetch use op_fetch_read instead of op_read (#7529)

ry committed:
fix: Use Buffer.writeSync in MultipartBuilder (#7542)

newtack commented:
Not sure this is the right place, but I would make the following change:

Batteries included: 
- Tooling: built-in Typescript compiling, testing, linting and formatting.
- Standard libraries replacing need for many 3rd party libraries

For me personally, another huge value add is that with the single executable I can upgrade the equivalent of Node, Typescript, eslint, jest, prettier in one command and all integrated.

Deno really removes a lot of friction.

lucacasonato committed:
chore: remove readJson/writeJson from manual (#7541)

bartlomieju committed:
refactor: disable URL.createObjectUrl (#7543)

bartlomieju committed:
refactor: move op_resources and op_close to deno_core (#7539)

Moves op_close and op_resources to deno_core::ops and exports them.
Adds serde dependency to deno_core and reexports it.

Moves JS implementation of those ops to Deno.core and reexports them in Deno.

bartlomieju committed:
refactor: use Symbol.for instead of Symbol in cli/rt/ (#7537)

caspervonb committed:
fix(cli/console): escape non printable characters in object entries (#7533)

bartlomieju committed:
refactor: Move URL to op_crates/web (#7544)

bartlomieju committed:
fix: disable rustyline logs (#7535)

g-plane committed:
fix(cli/fmt): make fmt output more readable (#7534)

JayHelton committed:
feat(std/node) stub out process.stdin, stdout, stderr (#7184)

bartlomieju committed:
refactor: deno_fetch op crate (#7524)

caspervonb committed:
fix(cli/console): always quote and escape inspected strings (#7546)

bartlomieju committed:
refactor: move FileReader to op_crates/web (#7554)

caspervonb committed:
fix(cli/console): surround non alpha-numeric object keys with quotes (#7550)

ry committed:
publish deno_fetch during CI (#7557)

caspervonb committed:
fix(cli/repl): format evaluation results with the object specifier (#7561)

tokiedokie committed:
fix(cli/fmt): canonicalize files in current dir (#7508)

ry committed:
Use reqwest from deno_fetch export (#7562)

nayeemrmn committed:
fix(bundle, eval, repl): Add missing flags (#7414)

Restructures flag helpers and applies them consistently.

TheWizardBear commented on README.md:
````diff
@@ -1,29 +1,86 @@
+
+Or a more complex one:
+
+```sh
````
I think this `sh` should be `js`.

ry commented:
@vitormmatos can you please check the CLA again - it's not green.

ry committed:
Merge branch 'master' into updated-readme

ry committed:
Merge branch 'master' into updated-readme

ry committed:
fmt

ry committed:
fmt

Skillz4Killz commented:
If Twitter is added perhaps Discord could be added? 🤔 

`[![Discord](https://img.shields.io/discord/684898665143206084?color=697EC4&label=Discord&logo=discord&logoColor=FDFEFE&style=flat-square)](https://discord.gg/deno)` You may need to enable the widget in discord server settings

vitormmatos commented:
> @vitormmatos can you please check the CLA again - it's not green.

Not sure what happened before, It was showing that I was signed in the CLA.
I tried it again just now and it's showing as signed.

<img width="1280" alt="Screen Shot 2020-09-20 at 22 37 18" src="https://user-images.githubusercontent.com/14911617/93727565-0c658c00-fb92-11ea-87ae-3bc48cbd5987.png">

kitsonk commented:
@vitormmatos the e-mail you used in the commit message does not match what you have registered on your GitHub account, and therefore it cannot identify you as the author of the commit:

```
commit 9e7966741d9a18a90dba62f8f58aac7bf5f8b105
Author: Vitor Matos <vitor@Vitors-MacBook-Pro.local>
Date:   Wed Sep 16 21:40:53 2020 -0300

    complete readme redesign
    
    updated deno description matching the official site.
    
    deno logo added aligned to the right side of the description.
    
    twitter badge added.
    
    installation, getting started and contributing sections have been added

```

There are ways to _hide_ your e-mail address with GitHub, under your profile settings, if you don't want it to be part of a commit message.  GitHub will issue you a specific GitHub e-mail address tied to your account.

vitormmatos commented:
> @vitormmatos the e-mail you used in the commit message does not match what you have registered on your GitHub account, and therefore it cannot identify you as the author of the commit:
> 
> ```
> commit 9e7966741d9a18a90dba62f8f58aac7bf5f8b105
> Author: Vitor Matos <vitor@Vitors-MacBook-Pro.local>
> Date:   Wed Sep 16 21:40:53 2020 -0300
> 
>     complete readme redesign
>     
>     updated deno description matching the official site.
>     
>     deno logo added aligned to the right side of the description.
>     
>     twitter badge added.
>     
>     installation, getting started and contributing sections have been added
> ```
> 
> There are ways to _hide_ your e-mail address with GitHub, under your profile settings, if you don't want it to be part of a commit message. GitHub will issue you a specific GitHub e-mail address tied to your account.

I have no problem with the e-mail been shown, but how can I fix it now?
I've changed the e-mail used on global config on my machine to match the one used on Github, what shall I do next for the CLA license to be properly signed. Should I do a new commit?

Sorry for the trouble.

Commit:
Merge branch 'updated-readme' of https://github.com/vitormmatos/deno into updated-readme

vitormmatos commented:
Just now I realized something quite strange. I'm signed and not signed in the CLA. My username is vitormmatos. Sorry for the trouble guys, really. Honestly, I have no idea what's going on here, and how to fix this.
<img width="123" alt="Screen Shot 2020-09-21 at 04 20 29" src="https://user-images.githubusercontent.com/14911617/93740967-e821a380-fbc1-11ea-93fe-8ef7107ab547.png">

hazae41 commented:
@vitormmatos I think you have to make a whole new branch, new commit and new PR

ry reviewed (approved):
LGTM - thanks @vitormmatos
